### ISSUE TITLE
General enhancements for ease of use

### ISSUE THREAD

**Issue by ManuelLR** [GeographicaGS/Docker-PostGIS#53]
- [ ] `PG_CONF` reloaded in every reboot
- [ ] Data folder permission fix in every reboot
- [ ] **BREAKING CHANGE** Use the same names in environment variables as the official postgres image.
- [ ] Default log only by stdout (I don't know what the current behaviour is)
- [ ] Ability to create additional databases and users by environment variables
- [ ] Auto-generate random password if one is not defined for the primary user. Perhaps an environment variable asking if you want to generate randomly or, if you do not define environment variable is created automatically (in this case would be necessary to see how we contemplate those cases in which the user does not want to define a password)
- [ ] Be able to specify a path to deposit sql files and preload them when the database is first started.

**Comment by larce-bhs** [comment]
>  Default log only by stdout (I don't know what the current behaviour is)

Take into account that postgres has a log configuration which logs all SQL queries (``log_statement = 'all'``), which I think is enabled by default.

This setting should be disabled by default, specially in production environments. It may be useful tough to in development environments.

**Comment by vehrka** [comment]
Add `unaccent` library from the contrib directory in order to be able to load as extension

**Comment by rbsolis** [comment]
tablefunc in order to pivot tables quickly.

**Comment by cayetanobv** [comment]
> > Default log only by stdout (I don't know what the current behaviour is)
> 
> Take into account that postgres has a log configuration which logs all SQL queries (`log_statement = 'all'`), which I think is enabled by default.
> 
> This setting should be disabled by default, specially in production environments. It may be useful tough to in development environments.

Maybe we could change `log_statement` default value to `ddl` in our Docker image.

**Comment by cayetanobv** [comment]
Regarding to compile contrib extensions in default image I have some doubts. An alternative could be to document in main README how to add a contrib extension to your project Dockerfile extending from base image. What do you think about it?

**Comment by vehrka** [comment]
It would be nice also to have a *vi* installed

**Comment by cayetanobv** [comment]
> It would be nice also to have a _vim_ installed and delightful to have a minimal .vimrc (https://gist.github.com/vehrka/71b289bfe79854da0d92e8480bd42f15)

I think is better if the people install Vim (or his choice text editor) in a custom or project derived image so we can maintain base image (this image) thinner.

**Comment by vehrka** [comment]
> > It would be nice also to have a _vim_ installed and delightful to have a minimal .vimrc (https://gist.github.com/vehrka/71b289bfe79854da0d92e8480bd42f15)
> 
> I think is better if the people install Vim (or his choice text editor) in a custom or project derived image so we can maintain base image (this image) thinner.

(edited my original coment)

Except when you deploy behind a firewalled environment and you cannot install anything.

In the image there is NO editor at all, nothing not even nano ...

**Comment by larce-bhs** [comment]
I agree with @cayetanobv, I think that it is not typical for docker images to have more than needed to run the target application/service in a context-less environment. This kind of software must be included in derived images.
